src/tasks/engine.ts

```diff
@@ -31,22 +59,59 @@ export class SafeGarboEngine extends BaseGarboEngine {
   }
 }
 
-export function runSafeGarboTasks(tasks: Task[]): void {
-  const engine = new SafeGarboEngine(tasks);
+export class BarfTaskEngine extends BaseGarboEngine<never, BarfTask> {
+  failures = 0;
 
-  try {
-    engine.run();
-  } finally {
-    engine.destruct();
+  available(task: BarfTask): boolean {
+    const validSobrieties = [Sobriety.EITHER, sober() ? Sobriety.SOBER : Sobriety.DRUNK];
+    return validSobrieties.includes(task.sobriety) && (task.ready === undefined || task.ready());
   }
-}
 
-export function runGarboTasks(tasks: Task[]): void {
-  const engine = new BaseGarboEngine(tasks);
+  execute(task: BarfTask): void {
+    tryFillLatte();
+    meatMood().execute(estimatedTurns());
+    safeRestore();
+
+    const startTurns = totalTurnsPlayed();
+
+    super.execute(task);
+
+    const foughtAnEmbezzler = get("lastEncounter") === "Knob Goblin Embezzler";
+    if (foughtAnEmbezzler) logEmbezzler(task.name);
+
+    if (
+      totalTurnsPlayed() > startTurns &&
+      !(typeof task.spendsTurn === "function" ? task.spendsTurn() : task.spendsTurn)
+    ) {
+      print(`We unexpectedly spent a turn doing ${task.name}!`, "red");
+    }
 
-  try {
-    engine.run();
-  } finally {
-    engine.destruct();
+    if (task.completed()) {
+      this.failures = 0;
+    } else {
+      this.failures += 1;
+      if (this.failures >= 3) {
+        throw new Error("Tried thrice to adventure, and failed each time. Aborting.");
+      }
+    }
+    postCombatActions();
   }
 }
+
+function runEngineTasks<A extends string, T extends Task<A>, E extends Engine<A, T>>(engineType: {
```

Comment:
god I love dumb typescript stuff, this gives me life.

Reply:
Should we have a meta engine eventually once eveything is grimoirized that itself runs all of our various sub-engines?